src/main/java/org/broadinstitute/hellbender/transformers/BQSRReadTransformer.java

```diff
@@ -101,75 +118,89 @@ public GATKRead apply(final GATKRead read) {
         if (emitOriginalQuals && ! read.hasAttribute(SAMTag.OQ.name())) { // Save the old qualities if the tag isn't already taken in the read
             try {
                 read.setAttribute(SAMTag.OQ.name(), SAMUtils.phredToFastq(read.getBaseQualities()));
-            } catch (IllegalArgumentException e) {
+            } catch (final IllegalArgumentException e) {
                 throw new UserException.MalformedRead(read, "illegal base quality encountered; " + e.getMessage());
             }
         }
 
-        final ReadCovariates readCovariates = RecalUtils.computeCovariates(read, header, covariates);
-        final int readLength = read.getLength();
+        final ReadCovariates readCovariates = RecalUtils.computeCovariates(read, header, covariates, false);
 
-        for (final EventType errorModel : EventType.values()) { // recalibrate all three quality strings
-            if (disableIndelQuals && errorModel != EventType.BASE_SUBSTITUTION) {
-                ReadUtils.setBaseQualities(read, null, errorModel);
-                continue;
-            }
+        //clear indel qualities
+        read.clearAttribute(ReadUtils.BQSR_BASE_INSERTION_QUALITIES);
+        read.clearAttribute(ReadUtils.BQSR_BASE_DELETION_QUALITIES);
+
+        // get the keyset for this base using the error model
+        final int[][] fullReadKeySet = readCovariates.getKeySet(EventType.BASE_SUBSTITUTION);
```

Comment:
Hopefully you see that if 1) we are hard coding the covariates and 2) are no longer doing indel recalibration, we no longer need these key sets and can be much less generic (and more efficient).  Hopefully that's the plan for a future optimization!